scripts/o3de/o3de/packaging.py

```diff
@@ -0,0 +1,318 @@
+        print(stderr.decode('ascii'))
+
+    return ret
+
+
+
+def package_none():
+    print("It seems like packaging is not supported for your platform or use case...")
+    return 0
+
+def package_hi():
+    print("We are using this function to say hi from another script file!")
+    return 0
+
+#this is a stand alone proof of concept test function, which just proves it's possible to automate building in Python
+#we will use the remainder of the code to work out the framework we would like to use to funnel building procedures
+#algorithm here is based on main documentation page for monolithic builds:
+#https://www.o3de.org/docs/user-guide/packaging/windows-release-builds/#debugging
+
+#QUESTIONS:
+#1) how do we go about detecting the correct asset processor batch executable? (do we fetch from vs_2022, or vs_2019?)
+#2) how do we formulaically specify the correct cmake generator?
+#3) what commands should we run to pre-build our executables? the o3de docs seem to indicate doing it from the project folder, but @stankowi provides commands from the engine folder
+#4) how do we standardize rules related to the packaging rules file found in the project? What parameters should go in there?
+
+#should we break this into smaller steps?
+# preliminary list:
+# step 1: validation - can we proceed?
+# step 2: preparation - asset processing, etc.
+# step 3: c++ build - actual build of c++ code with config
+# step 4: bundle content for use
+# step 5: prep non c++/non asset content 
+# step 6: prep directory layout with all prepared content
+# step 7: backup then post validation
+# step 8: putting everything into container for distribution (zip file, apk file, ipa file)
+#enable code re-use. This could also inform _how_ we setup our keys
+
+# we can rely on making this code file expandable as a fallback
+# we should provide legacy warnings as code updates progress
+
+# how do we handle with engine customer/contributor
+# RFC will help inform my logistical questions on what needs to be split and when
+# test driven development would help
+def package_windows_standalone_monolithic(project_path: str,
+                                          zipped: bool):
+    result = 0
+
+    engine_path = manifest.get_project_engine_path(pathlib.Path(project_path))
+
+    #packaging rules should be in a standard location with respect to the project. Note, if this is not found, we need to have reasonable defaults
+    PROJECT_PATH = project_path
+    PACKAGING_RULES_PATH = os.path.join(PROJECT_PATH, "packaging",  "packaging_config.packagerules")
+
+    packaging_rules = None
+    with open(PACKAGING_RULES_PATH, "r") as prules_file:
+        packaging_rules = json.load(prules_file)
+        print(packaging_rules)
+
+    #note: we need ability to supply a default option!
+    ENGINE_PATH = packaging_rules["engine-path"]
+    THIRD_PARTY_PATH  = packaging_rules["third-party-path"]
+    CMAKE_GENERATOR = packaging_rules["cmake-generator"]
```

Comment:
Feedback from our review: Is there an existing cmake python interface we use elsewhere, that we can re-use here?

Reply:
What about https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/cmake.py ?